src/main/java/com/echobox/cache/CacheWithSupplierFailover.java

```diff
@@ -18,20 +18,28 @@
 package com.echobox.cache;
 
 import com.google.gson.reflect.TypeToken;
-import org.apache.commons.lang3.NotImplementedException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.io.Serializable;
+import java.util.Objects;
 import java.util.function.Supplier;
 
 /**
- * Periodically executes the callback to getWithFailover the latest value.
- * If it's within the specified period then getWithFailover the cached value.
- * If the callback fails still getWithFailover the cached value and update the cache next time
+ * The cache that periodically gets the latest value from the provided source of truth.
+ * If source of truth is not available then it will use
+ * cached value for a specified period of time before throwing an error.
  * @param <T> return type of cache object
  * @author Daniyar
  */
 public class CacheWithSupplierFailover<T extends Serializable> {
   
+  private static final Logger logger = LoggerFactory.getLogger(CacheWithSupplierFailover.class);
+  
+  private final String defaultCacheKeySuffix = "-default";
+  
+  private final String errorCacheKeySuffix = "-error";
```

Comment:
These strings can be static and I think the naming convention is that the field names should be capitalised with `_` for spaces.
```suggestion
  private static final String DEFAULT_CACHE_KEY_SUFFIX = "-default";
  
  private static final String ERROR_CACHE_KEY_SUFFIX = "-error";
```